src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -116,17 +115,15 @@ string stripAndLowercaseKeywordLiteral(std::string_view lit) {
 template <typename Current, typename... Others>
 constexpr const ad_utility::Last<Current, Others...>* unwrapVariant(
     const auto& arg) {
-  if constexpr (sizeof...(Others) > 0) {
-    if constexpr (ad_utility::isSimilar<decltype(arg), Current>) {
-      if (const auto ptr = std::get_if<ad_utility::First<Others...>>(&arg)) {
-        return unwrapVariant<Others...>(*ptr);
-      }
-      return nullptr;
-    } else {
-      return unwrapVariant<Others...>(arg);
+  if constexpr (sizeof...(Others) == 0) {
+    return &arg;
+  } else if constexpr (ad_utility::isSimilar<decltype(arg), Current>) {
+    if (const auto ptr = std::get_if<ad_utility::First<Others...>>(&arg)) {
+      return unwrapVariant<Others...>(*ptr);
     }
+    return nullptr;
   } else {
-    return &arg;
+    return unwrapVariant<Others...>(arg);
```

Comment:
Just curious: what was the code smell here in the previous version of the code?

Reply:
The code smell here was that there was a nested `if` that could also be expressed with a single layer of `if-else if` statements without the nesting.

The actual code smell of course here is that this is a very useful function for some parsing contexts, but it is rather complicated (a lot of template meta stuff) and not documented.
I will write something for it.